src/parser/esprima_cpp/esprima.cpp

```diff
@@ -5562,20 +5608,21 @@ class Parser : public gc {
         this->consumeSemicolon();
 
         if (isLiteral) {
-            return this->finalize(node, new DirectiveNode(asExpressionNode(expr).get(), directiveValue));
+            return this->finalize(node, new DirectiveNode(asExpressionNode(expr).get(), *directiveValue));
```

Comment:
before return, check nullity of ``new`` with ASSERT

Reply:
ASTNode is also allocated by GC. So, if there is any problem in allocation, GC will notify us.